python/lsst/multiprofit/sourceconfig.py

```diff
@@ -0,0 +1,385 @@
+        if errors:
+            newline = "\n"
+            raise ValueError(f"ComponentMixtureConfig has validation errors:\n{newline.join(errors)}")
+
+
+class SourceConfig(pexConfig.Config):
+    """Configuration for a gauss2d.fit Source.
+
+    Sources may contain components with distinct centroids that may be linked
+    by a prior (e.g. a galaxy + AGN + star clusters),
+    although such priors are not yet implemented.
+    """
+
+    component_groups = pexConfig.ConfigDictField[str, ComponentGroupConfig](
+        doc="Components in the source",
+        optional=False,
+    )
+
+    def _make_components_priors(
+        self,
+        component_group_fluxes: list[list[Fluxes]],
+        label_integral: str,
+        validate_psf: bool = False,
+    ) -> [list[g2f.Component], list[g2f.Prior]]:
+        if len(component_group_fluxes) != len(self.component_groups):
+            raise ValueError(f"{len(component_group_fluxes)=} != {len(self.component_groups)=}")
+        components = []
+        priors = []
+        if validate_psf:
+            keys_expected = tuple((g2f.Channel.NONE,))
+        for component_fluxes, (name_group, component_group) in zip(
+            component_group_fluxes, self.component_groups.items()
+        ):
+            if validate_psf:
+                for idx, fluxes_comp in enumerate(component_fluxes):
+                    keys = tuple(fluxes_comp.keys())
+                    if keys != keys_expected:
+                        raise ValueError(
+                            f"{name_group=} comp[{idx}] {keys=} != {keys_expected=} with {validate_psf=}"
+                        )
+
+            components_i, priors_i = component_group.make_components(
+                component_fluxes=component_fluxes,
+                label_integral=self.format_label(label=label_integral, name_group=name_group),
+            )
+            components.extend(components_i)
+            priors.extend(priors_i)
+
+        return components, priors
+
+    @staticmethod
+    def format_label(label: str, name_group: str) -> str:
+        return string.Template(label).safe_substitute(name_group=name_group)
+
+    def get_component_configs(self) -> ComponentConfigs:
+        has_prefix_group = self.has_prefix_group()
+        component_configs = {}
+        for name_group, config_group in self.component_groups.items():
+            prefix_group = f"{name_group}_" if has_prefix_group else ""
+            for name_comp, component_config in config_group.get_component_configs().items():
+                component_configs[f"{prefix_group}{name_comp}"] = component_config
+        return component_configs
+
+    def get_integral_label_default(self) -> str:
+        prefix = "mix: ${name_group} " if self.has_prefix_group() else ""
+        return f"{prefix}{ComponentGroupConfig.get_integral_label_default()}"
+
+    def has_prefix_group(self) -> bool:
+        return (len(self.component_groups) > 1) or next(iter(self.component_groups.keys()))
```

Comment:
While `next(iter(self.component_groups.keys())` is truthy, I would recommend wrapping it in a `bool` so that the return type matches the type hint.

Reply:
Curiously, mypy didn't complain about this.